doc/man/man3/crypto_argon2.3monocypher

```diff
@@ -132,30 +111,92 @@ If you are not sure how to allocate that buffer, just use
 .Fn malloc .
 .Pp
 The work area is automatically wiped by
-.Fn crypto_argon2i .
+.Fn crypto_argon2 .
+.It Fa config
+A struct of type
+.Vt crypto_argon2_config
+that determines the base parameters of this particular
+instance of Argon2.
+These are domain parameters and remain constant between
+multiple invocations of
+.Fn crypto_argon2 .
+.It Fa inputs
+A struct of type
+.Vt crypto_argon2_inputs
+that contains the actual input parameters.
+.It Fa extras
+A struct of type
+.Vt crypto_argon2_extras
+that contains optional extra input parameters,
+which are not commonly used.
+.El
+.Pp
+The
+.Vt crypto_argon2_config
+struct is defined as follows:
+.Bd -literal -offset indent
+typedef struct {
+    uint32_t algorithm;
+    uint32_t nb_blocks;
+    uint32_t nb_passes;
+    uint32_t nb_lanes;
+} crypto_argon2_config;
+.Ed
+.Pp
+Its members are:
+.Bl -tag -width Ds
+.It Fa algorithm
+This value determines which variant of Argon2 should be used.
+.Dv CRYPTO_ARGON2_D
+indicates Argon2d,
+.Dv CRYPTO_ARGON2_I
+indicates Argon2i,
+.Dv CRYPTO_ARGON2_ID
+indicates Argon2id.
 .It Fa nb_blocks
 The number of blocks for the work area.
 Must be at least 8.
 A value of 100000 (one hundred megabytes) is a good starting point.
 If the computation takes too long, reduce this number.
 If it is too fast, increase this number.
 If it is still too fast with all available memory, increase
-.Fa nb_iterations .
-.It Fa nb_iterations
-The number of iterations.
+.Fa nb_passes .
+.It Fa nb_passes
+The number of passes.
 It must be at least 1.
 A value of 3 is
 .Em strongly
-recommended;
+recommended when using Argon2i;
 any value lower than 3 enables significantly more efficient attacks.
-.It Fa password
+.It Fa nb_lanes
+The level of parallelism.
+This parameter has no effect as Monocypher has no concept of threading.
```

Comment:
Strictly speaking, this parameter *does* have an effect, since the resulting hash will conform to what the Argon2 specification says. It's just that:

- Even though it _could_ be computed in parallel with at most `nb_lanes` threads, Monocypher uses only 1 thread.
- Therefore, using any value other than 1 has the very real effect of increasing attacker advantage, effectively _reducing_ security.

By the way, I recall having read a paper suggesting that users who want actual parallelism should instead run several instances of Argon2 in parallel themselves, (using different additional data for each thread), then hash the results together in one final hash.

Reply:
I have now modified it to my satisfaction. Further corrections are welcome if I messed it up though.